aidial_adapter_vertexai/chat/gemini/adapter.py

```diff
@@ -242,6 +245,28 @@ async def set_finish_reason(
         await consumer.set_finish_reason(openai_reason)
 
 
+async def create_attachments_from_citations(
+    response: GenerationResponse,
+    consumer: Consumer,
+    citations_limit: int = DEFAULT_CITATIONS_LIMIT,
+) -> None:
+    if response.candidates is None or not len(response.candidates):
+        return None
+    citation_metadata = response.candidates[0].citation_metadata
+
+    if (
+        citation_metadata is None
+        or citation_metadata.citations is None
+        or not len(citation_metadata.citations)
+    ):
+        return None
+
+    for citation in citation_metadata.citations[:citations_limit]:
+        await consumer.add_attachment(
```

Comment:
Make sure that `citation.uri` isn't None too.
I see the following error:

```
pydantic.error_wrappers.ValidationError: 1 validation error for AttachmentChunk
__root__
  Trying to add attachment without data and url (type=value_error)
```

when I send the following request: `what are capitals of Latin America?`

Reply:
Yeah, interesting case
I haven't met this on just `gemini-pro`, but on `gemini1.5-pro` it reproduced

Somehow that citation had title, but uri was empty
Thanks for noticing!
---
Fixed